src/custom_request_schedule_manager.h

```diff
@@ -80,8 +79,18 @@ class CustomRequestScheduleManager : public RequestRateManager {
       const std::shared_ptr<ModelParser>& parser,
       const std::shared_ptr<cb::ClientBackendFactory>& factory);
 
-  /// Generates and updates the request schedule as per the given schedule
-  void GenerateSchedule();
+  void DistributeScheduleToWorkers(const Schedule& schedule);
+
+ private:
+  void InitManagerFinalize() override;
+
+  std::pair<Schedule, Schedule> GetSchedulesFromDataset() const;
+
+  std::chrono::milliseconds GetTimestamp(size_t dataset_index) const;
+
+  void PerformWarmup();
+
+  void InitCustomSchedule(const Schedule& schedule, size_t dataset_offset = 0);
 
   /// Creates worker schedules based on the provided schedule
   /// \param duration The maximum duration for the schedule
```

Comment:
I know you're not touching this, but... we should remove this line as it's not actually true.

Reply:
Removed 👍